src/main/java/dev/openfeature/javasdk/EvaluationContext.java

```diff
@@ -33,16 +17,18 @@ public EvaluationContext(String targetingKey) {
      * @param ctx2 overriding context
      * @return resulting merged context
      */
-    public static EvaluationContext merge(EvaluationContext ctx1, EvaluationContext ctx2) {
-        EvaluationContext ec = new EvaluationContext();
+    static EvaluationContext merge(EvaluationContext ctx1, EvaluationContext ctx2) {
         if (ctx1 == null) {
             return ctx2;
         } else if (ctx2 == null) {
             return ctx1;
         }
 
-        ec.structure.attributes.putAll(ctx1.structure.attributes);
-        ec.structure.attributes.putAll(ctx2.structure.attributes);
+        Map<String, Value> merged = new HashMap<String, Value>();
+
+        merged.putAll(ctx1.asMap());
+        merged.putAll(ctx2.asMap());
+        EvaluationContext ec = new MutableContext(merged);
```

Comment:
My java knowledge is extremely limited but it's surprising to me that there's a concrete type being returned in the interface definition

Reply:
I went for the approach of including the merge as a method in the `EvaluationContext` interface, thereby leaving it to each implementation:
https://github.com/open-feature/go-sdk/pull/92/files#diff-d0bdacefeb6c765ab5d7987bdfcd09508b4f76e591bc7720f1a484c836625140R15

Reply:
I've addressed this as above. Thanks! (now you have to help write Java)